teps/0138-decouple-api-and-feature-versioning.md

```diff
@@ -72,8 +91,330 @@ Per [TEP-0033](https://github.com/tektoncd/community/blob/main/teps/0033-tekton-
+
+Note that although this looks like a behavior change, it is actually a bug fix. Currently, with `enable-api-fields` set to `stable`,  PipelineRuns like [this one](https://github.com/tektoncd/pipeline/blob/main/examples/`v1`/pipelineruns/beta/git-resolver.yaml) fail because the controller cannot create child TaskRuns. This change will result in a validation failure instead and this `pipelineSpec` will be prohibited with ``enable-api-fields=stable`.
+
+However, the [default](https://github.com/tektoncd/pipeline/blob/main/config/config-feature-flags.yaml#L89) value of `enable-api-fields` continues to be `beta`. The default deployment of Tekton Pipelines comes with the current [beta features](https://github.com/tektoncd/pipeline/blob/main/docs/additional-configs.md#beta-features) enabled.
+
+- **Impacts on users:**
+    - Cluster Operators:
+      - No action needed from the cluster operator. The existing deployments with `enable-api-fields` set to `alpha` or `beta` should not experience any changes.
+      - This makes it possible for the cluster operators to have a full control over their deployments. The cluster operator can guarantee enabling only `stable` features and avoid user bypassing this enforcement . For example, currently it is not possible for the cluster operators to enable only `stable` features for `v1beta1` with an exception of a list of beta features such as resolver, object params and results.
+    - Pipeline and Task Authors: 
+      - The same `pipeline` and `task` definitions might not be applied to a cluster with `enable-api-fields` set to `stable` after this change is implemented.
+      - The `pipeline` and `task` definitions in `v1beta1` implementing [beta features](https://github.com/tektoncd/pipeline/blob/main/docs/additional-configs.md#beta-features) will not be applied to a cluster with `enable-api-fields` set to `stable` Such pipeline with beta features used in `v1beta1` apiVersion would not result in a successful pipelineRun because of [the validation after the conversion to `v1` as the storage version](https://github.com/tektoncd/pipeline/blob/d9d2d1760fa534e2dfb16ca656cdf2d293a5900e/pkg/apis/pipeline/`v1`/taskref_validation.go#L39). Since in this example, resolver is still a beta feature, this pipeline should not have been applied to the cluster that has not set `enable-api-fields=beta` in the first place.
+
+### Per feature flag for _new_ api-driven features
```

Comment:
It seems like the API versions and feature stability levels are already decoupled with the [Change existing validation to decouple feature and API versioning](https://github.com/tektoncd/community/blob/a5462c5dc565883646f8c6b719c07f046fa34999/teps/0138-decouple-api-and-feature-versioning.md#change-existing-validation-to-decouple-feature-and-api-versioning) proposal. I'm wondering what is the motivation to use per-feature flag model?

Reply:
Thanks for the comment. I've put back the missing pieces:)

The reason that this was not previously included was that we first aimed to decouple feature versioning only and while moving forward with the discussion in API WG we've decided to use per-feature flag as the solution to decouple versioning in order to avoid the potential issues from setting the default stability level in the long run.

Reply:
Thanks!